src/schema-association-service.ts

```diff
@@ -98,13 +98,16 @@ async function autoDetectSchema(
     workspaceFolder: vscode.WorkspaceFolder): Promise<vscode.Uri | undefined> {
     const azureAccountApi = await getAzureAccountExtensionApi();
 
+    const doNotAskAgainSignIn = context.globalState.get<boolean>('doNotAskAgainSignIn');
+    logger.log(`doNotAskAgainSignIn status: ${doNotAskAgainSignIn}`);
+
     // We could care less about the subscriptions; all we need are the sessions.
     // However, there's no waitForSessions API, and waitForLogin returns before
     // the underlying account information is guaranteed to finish loading.
     // The next-best option is then waitForSubscriptions which, by definition,
     // can't return until the sessions are also available.
     // This only returns false if there is no login.
-    if (!(await azureAccountApi.waitForSubscriptions())) {
+    if (!(await azureAccountApi.waitForSubscriptions()) && !doNotAskAgainSignIn) {
```

Comment:
Move this into its own separate check, like you did on L188

Reply:
Moved above